src/main/java/org/broadinstitute/hellbender/tools/walkers/genotyper/AlleleSubsettingUtils.java

```diff
@@ -472,11 +472,22 @@ public static int[] getIndexesOfRelevantAllelesForGVCF(final List<Allele> remapp
         // create the index mapping, using the <NON-REF> allele whenever such a mapping doesn't exist
         for ( int i = 1; i < targetAlleles.size(); i++ ) {
             // if there's more than 1 spanning deletion (*) allele then we need to use the best one
-            if (targetAlleles.get(i) == Allele.SPAN_DEL && !doSomaticMerge && g.hasPL()) {
+            if (targetAlleles.get(i) == Allele.SPAN_DEL) {
                 final int occurrences = Collections.frequency(remappedAlleles, Allele.SPAN_DEL);
                 if (occurrences > 1) {
-                    final int indexOfBestDel = indexOfBestDel(remappedAlleles, g.getPL(), g.getPloidy());
-                    indexMapping[i] = (indexOfBestDel == -1 ? indexOfNonRef : indexOfBestDel);
+                    // If we have PLs and it's not a somatic merge, use the best spanning deletion based on PLs
```

Comment:
Instead of splitting the single vs multiple `SPAN_DEL` cases, can this all be reduced to a stream one-liner of finding the maximum?